### ISSUE TITLE
mnist_cnn.py major performance loss after switching to TF 1.14

### ISSUE THREAD

**Issue by mikejcroc** [tensorflow/tensorflow#31905]
**System information**
- Windows 10 Enterprise 64-bit
- TensorFlow GPU installed via Anaconda
- TensorFlow GPU version: 1.14
- Python version: 3.7.3
- CUDA version: 10.0.130
- cuDNN version: 7.6.0
- GPU model and memory: NVIDIA Quadro P600 (2 GB)

**Describe the current behavior**
The mnist.py script achieves a test accuracy of <85% after 12 epochs of training when using tensorflow-gpu=1.14.

**Describe the expected behavior**
The mnist.py script is supposed to to achieve >99% test accuracy after 12 epochs of training. Doing a clean install of tensorflow-gpu=1.13 and running the script achieves this result.

- TensorFlow GPU version: 1.13
- Python version: 3.7.3
- CUDA version: 10.0.130
- cuDNN version: 7.6.0

This discrepancy between versions exists even when setting the numpy and TensorFlow random seeds before training. There are potentially major implications for the reproducibility of any work done in TensorFlow across these versions. Why is this happening and/or how can it be fixed?

**Code to reproduce the issue**
Use [mnist.py](https://github.com/keras-team/keras/blob/master/examples/mnist_cnn.py) and replace keras with tf.keras

**Other info / logs**
None.

**Comment by vlawhern** [comment]
This issue appears to be due to a regression in the 'Adadelta' optimizer in TF1.14. I tested this with TF1.13 and 1.14 with Adadelta and Adam and noticed the following:

In TF1.13 with Adadelta = script works as intended
in TF1.14 with Adadelta = script doesn't work
in TF1.14 with Adam = script works as intended

**Comment by vlawhern** [comment]
Upon further investigation this is due to a change in the default learning rate in Adadelta in TF1.14, see Issue #31024. Changing the learning rate to 1.0 when in TF1.14 fixes the issue.

**Comment by mikejcroc** [comment]
Thanks for looking into this! That seems to be the issue.
